tachyon/math/polynomials/univariate/univariate_sparse_coefficients.h

```diff
@@ -104,13 +104,13 @@ class UnivariateSparseCoefficients {
   constexpr explicit UnivariateSparseCoefficients(
       const std::vector<Term>& terms)
       : terms_(terms) {
+    RemoveZeros();
     CHECK_LE(Degree(), kMaxDegree);
-    RemoveHighDegreeZeros();
```

Comment:
Why did you switch the order?

Reply:
It will handle cases when the highest degree terms are 0.

Reply:
To rephrase, Degree() can become less that kMaxDegree after RemoveZeros()